Fatras/include/ActsFatras/Kernel/detail/Interactor.hpp

```diff
@@ -69,45 +77,79 @@ struct Interactor {
                   result_type &result) const {
     assert(generator and "The generator pointer must be valid");
 
+    // arm the point-like interactions before the first step
+    if (std::isnan(result.properTime)) {
+      arm(initialParticle, result);
+    }
```

Comment:
No, because the `Interactor` (or more specifically the operator that is called by the propagator) must be `const`. The X0/L0 limit must be assignable for the reasons described above. The only object that is modifiable during the call and that we have control over is the result object.